tools/write_release_and_log.py

```diff
@@ -0,0 +1,145 @@
+    else:
+        GIT_REVISION = "Unknown"
+
+    if not version_utils.ISRELEASED:
+        if GIT_REVISION == "Unknown":
+            FULLVERSION += '.dev0+Unknown'
+        else:
+            FULLVERSION += '.dev0+' + GIT_REVISION[:7]
+finally:
+    sys.path.pop(1)
+    sys.path.pop(0)
+
+try:
+    # Ensure sensible file permissions
+    os.umask(0o022)
+except AttributeError:
+    # No umask on non-posix
+    pass
+
+
+def get_latest_release_doc(path):
+    """
+    Method to Pick the latest file from 'doc/release'
+    """
+    files_path = os.path.join(path, '*')
+    files = sorted(
+        glob.iglob(files_path), key=os.path.getmtime, reverse=True)
+    return files[0]
+
+# ----------------------------
+# Release notes and Changelog
+# ----------------------------
+
+
+def compute_md5(idirs):
+    released = os.listdir(idirs)
+    checksums = []
+    for fn in sorted(released):
+        fn_updated = os.path.join("release", fn)
+        with open(fn_updated, 'rb') as f:
+            m = md5(f.read())
+        checksums.append('%s  %s' % (m.hexdigest(), os.path.basename(fn)))
+    return checksums
+
+
+def compute_sha256(idirs):
+    # better checksum so gpg signed README.txt containing the sums can be used
+    # to verify the binaries instead of signing all binaries
+    released = os.listdir(idirs)
+    checksums = []
+    for fn in sorted(released):
+        fn_updated = os.path.join("release", fn)
+        with open(fn_updated, 'rb') as f:
+            m = sha256(f.read())
+        checksums.append('%s  %s' % (m.hexdigest(), os.path.basename(fn)))
+
+    return checksums
+
+
+def write_release_task(filename='NOTES.txt'):
+    idirs = Path('release')
+    source = Path(RELEASE)
+    target = Path(filename)
+    if target.exists():
+        target.remove()
+
+    tmp_target = Path(filename + '.txt')
+    os.system(f'cp {source} {tmp_target}')
+
+    with open(str(tmp_target), 'a') as ftarget:
+        ftarget.writelines("""
+Checksums
+=========
+
+MD5
+~~~
+
+""")
+        ftarget.writelines(['%s\n' % c for c in compute_md5(idirs)])
+        ftarget.writelines("""
+SHA256
+~~~~~~
+
+""")
+        ftarget.writelines(['%s\n' % c for c in compute_sha256(idirs)])
+
+
+def write_log_task(filename='Changelog'):
+    st = subprocess.Popen(
+        ['git', 'log', '%s..%s' % (LOG_START, LOG_END)],
+        stdout=subprocess.PIPE)
+
+    out = st.communicate()[0].decode()
+    with open(filename, 'w') as a:
+        a.writelines(out)
+
+
+def main():
+    """
+    Checks weather release directory is present or not
+    and calls the method to generate logs and notes
+    """
+    if not os.path.exists("release"):
+        os.makedirs("release")
+    else:
+        print('Release directory present, executing release tasks')
+    write_release_task(os.path.join("release", 'README'))
+    write_log_task(os.path.join("release", 'Changelog'))
+    print("Release Logs and Readme generated successfully")
```

Comment:
I suppose `README` should be capitalized. I noticed that the new script produces `README.txt` vs. the old approach producing `README` with no extension. I don't really mind either way--I kind of end up pasting it in 4 different places regardless. I guess it may be restructured text (`rst`) content, which I then somewhat-reformat to markdown manually for the GitHub release page.